### PR TITLE
[Yaml] throw meaningful exception for complex mappings

### PR DESCRIPTION
**Pull request by xabbuh**
| Q             | A
| ------------- | ---
| Branch?       | master
| Bug fix?      | no
| New feature?  | no
| BC breaks?    | no
| Deprecations? | no
| Tests pass?   | yes
| Fixed tickets | #20579
| License       | MIT
| Doc PR        |

### CODE REVIEW

**Comment by xabbuh** [comment]
In fact, I think we need to deprecate the current behaviour in 3.3 first and merge this one for 4.0 (see #22059 for the deprecation).

File: src/Symfony/Component/Yaml/Parser.php [hunk 1]
```diff
@@ -144,6 +144,10 @@ public function parse($value, $flags = 0)
                     $values['value'] = $matches['value'];
                 }
 
+                if (isset($values['value'][1]) && '?' === $values['value'][0] && ' ' === $values['value'][1]) {
+                    throw new ParseException('The parser does not support complex mappings.', $this->currentLineNb + 1, $this->currentLine);
```
Comment (javiereguiluz) [inline src/Symfony/Component/Yaml/Parser.php]: `Complex mappings are not supported.` ? To match the style of other error messages (e.g. `'Multiple documents are not supported.'`)

**Review by javiereguiluz — APPROVED** [review]
👍

**Review by nicolas-grekas — APPROVED** [review]
:+1:

**Comment by xabbuh** [comment]
For the records: This change is part of #22770.

**Pull request closed without merging**